src/robotFactory.js

```diff
@@ -1,10 +1,88 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {
+    x: 0, y: 0,
+  }, chipVersion) {
+    if (!coords.hasOwnProperty('x')) {
+      coords.x = 0;
+    };
 
-class FlyingRobot {}
+    if (!coords.hasOwnProperty('y')) {
+      coords.y = 0;
+    };
 
-class DeliveryDrone {}
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
```

Comment:
```suggestion
        this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```

And you can remove your checks above